File: leetcode/0217-contains-duplicate/solution.cpp
class Solution {
public:
    /**
     * RUNTIME: 90 ms (97.85%)
     * MEMORY: 51.3 MB (70.17%)
    */

    bool containsDuplicate(vector<int>& nums) {
        if (nums.empty()) { return false; }
        unordered_map<int,int> hashset;
        for (int i : nums) {
            if (++hashset[i] > 1) {
                return true;
            }
        }
        return false;
    }
};

File: leetcode/0217-contains-duplicate/solution.py
class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        """
        RUNTIME: 594 ms (75.2%)
        MEMORY: 26 MB (28.51%)
        """
        if len(nums) == 0: return False
        hashset = {}
        for i in nums:
            if i not in hashset:
                hashset[i] = 1
            else:
                return True
        return False
